crates/ironrdp-server/src/encoder/mod.rs

```diff
@@ -85,11 +83,11 @@ impl UpdateEncoder {
             xor_bpp: 32,
             color_pointer,
         };
-        let buf = self.pdu_encoder.encode(ptr)?;
-        Ok(UpdateFragmenter::new(UpdateCode::NewPointer, buf))
+        Ok(UpdateFragmenter::new(UpdateCode::NewPointer, encode_vec(&ptr)?))
     }
 
-    pub(crate) fn color_pointer(&mut self, ptr: ColorPointer) -> Result<UpdateFragmenter<'_>> {
+    #[allow(clippy::unused_self)]
```

Comment:
nitpick: I think it’s clearer to define the method as an associated function or a free standing function instead of an instance method if it doesn’t require self. Exception: avoid API breaking changes in the future ("maybe self will be used in the future"). But this API is private so it’s not a concern.